### PR TITLE
Python: Shared dataflow, conditionals

### PR DESCRIPTION
**Pull request by yoff**
This PR adds flow from the operands of `IfExprNode`s to the nodes themselves. It leans on the logic already implemented to exclude false positives in simple cases.

The dataflow step is simply added to `essaFlowStep`; we should perhaps rename that predicate...

Since I added new test cases to demonstrate that some false positives do not appear, I also formatted the test file with my new formatter of choice (black). This means that many line numbers are shifted, but the upside is that comments are closer to what they comment than to the definition above rather than the other way around. It also saves som lines overall.

Reading the commits in order might help, since all line shifting happens on the first commit.

### CODE REVIEW

File: python/ql/test/experimental/dataflow/coverage/test.py [hunk 1]
```diff
@@ -77,16 +75,14 @@ def test_imagnumber_literal():
...
     SINK(x)  # Flow missing
 
-# 6.2.3. Parenthesized forms
-
 
+# 6.2.3. Parenthesized forms
 def test_parenthesized_form():
-    x = (SOURCE)
+    x = SOURCE
     SINK(x)
```
Comment (RasmusWL) [inline python/ql/test/experimental/dataflow/coverage/test.py:82]: with autoformatting, this test suddently became trivial :sweat_smile:  :disappointed:

File: python/ql/test/experimental/dataflow/coverage/test.py [hunk 2]
```diff
@@ -77,16 +75,14 @@ def test_imagnumber_literal():
     x = 42j
     SINK(x)  # Flow missing
 
-# 6.2.3. Parenthesized forms
-
 
+# 6.2.3. Parenthesized forms
 def test_parenthesized_form():
-    x = (SOURCE)
+    x = SOURCE
```
Comment (calumgrant) [inline python/ql/test/experimental/dataflow/coverage/test.py:81]: Did the formatter get a bit aggressive here?

**Review by RasmusWL — CHANGES_REQUESTED** [review]
It would be super nice if you always put autoformatting commits in a separate commit, since it makes things a bit easier to scan. Now the title of 8f13d586b7662f39f60e9613aebeec511ebb861e _Python: More tests of conditonals_ lead me to believe that there would be new test-cases, but after looking through the diff of the test.py file 3 times, I haven't found any. Maybe just confusing commit message? (at least consider me confused)

Otherwise, I think we need to restore the parenthesis test. code-formatters are pretty nice, but sometimes it just doesn't do what we want :( besides this, change LGTM :+1:

File: python/ql/test/experimental/dataflow/coverage/test.py (hunk 2 above)
Comment (yoff) [reply python/ql/test/experimental/dataflow/coverage/test.py:81]: Hah, probably. I missed that, thanks!

File: python/ql/test/experimental/dataflow/coverage/test.py (hunk 1 above)
Comment (yoff) [reply python/ql/test/experimental/dataflow/coverage/test.py:82]: There are two new tests in there: https://github.com/github/codeql/commit/8f13d586b7662f39f60e9613aebeec511ebb861e#diff-601e823041b1218704b390c8387e5bb6R412-R426
I agree that they are near impossible to spot in that view though. Doing the autoformatting on its own is the right way, thanks for that suggestion. Would you like me to rebase to do that?

**Comment by yoff** [comment]
There are two new tests in there: https://github.com/github/codeql/commit/8f13d586b7662f39f60e9613aebeec511ebb861e#diff-601e823041b1218704b390c8387e5bb6R412-R426
I agree that they are near impossible to spot in that view though. Doing the autoformatting on its own is the right way, thanks for that suggestion. Would you like me to rebase to do that?

File: python/ql/test/experimental/dataflow/coverage/test.py (hunk 1 above)
Comment (yoff) [reply python/ql/test/experimental/dataflow/coverage/test.py:82]: Yes, autoformatting is dangerous like that... 😥

**Comment by RasmusWL** [comment]
> I agree that they are near impossible to spot in that view though. Doing the autoformatting on its own is the right way, thanks for that suggestion. Would you like me to rebase to do that?

Nah, lets just keep it in mind moving forwards :+1:

**Review by RasmusWL — APPROVED** [review]

**Pull request merged**